Sources/IRGen/TypeIRGen.swift

```diff
@@ -18,7 +18,9 @@ extension IRGenerator {
     }
     let structure = builder.createStruct(name: expr.name.name)
     typeIRBindings[expr.type] = structure
-    let fieldTypes = expr.properties.map { resolveLLVMType($0.type) }
+    let fieldTypes = expr.properties
```

Comment:
Maybe we want to extract this out into a "concreteStorageProperties" or something?